tests/components/knx/test_config_flow.py

```diff
@@ -0,0 +1,338 @@
+        assert result["type"] == RESULT_TYPE_FORM
+        assert not result["errors"]
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_KNX_CONNECTION_TYPE: CONF_KNX_TUNNELING,
+        },
+    )
+    await hass.async_block_till_done()
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["step_id"] == "manual_tunnel"
+    assert not result2["errors"]
+
+    with patch(
+        "homeassistant.components.knx.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result3 = await hass.config_entries.flow.async_configure(
+            result2["flow_id"],
+            {
+                CONF_HOST: "192.168.0.1",
+                CONF_PORT: 3675,
+            },
+        )
+        await hass.async_block_till_done()
+        assert result3["type"] == RESULT_TYPE_CREATE_ENTRY
+        assert result3["title"] == "192.168.0.1"
+        assert result3["data"] == {
+            CONF_KNX_CONNECTION_TYPE: CONF_KNX_TUNNELING,
+            CONF_HOST: "192.168.0.1",
+            CONF_PORT: 3675,
+            CONF_KNX_INDIVIDUAL_ADDRESS: "15.15.250",
+            ConnectionSchema.CONF_KNX_ROUTE_BACK: False,
+        }
+
+        assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_tunneling_setup_for_multiple_found_gateways(hass: HomeAssistant) -> None:
+    """Test tunneling if only one gateway is found."""
+    gateway = _gateway_descriptor("192.168.0.1", 3675)
+    gateway2 = _gateway_descriptor("192.168.1.100", 3675)
+    with patch("xknx.io.gateway_scanner.GatewayScanner.scan") as gateways:
+        gateways.return_value = [gateway, gateway2]
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert not result["errors"]
+
+    tunnel_flow = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_KNX_CONNECTION_TYPE: CONF_KNX_TUNNELING,
+        },
+    )
+    await hass.async_block_till_done()
+    assert tunnel_flow["type"] == RESULT_TYPE_FORM
+    assert tunnel_flow["step_id"] == "tunnel"
+    assert not tunnel_flow["errors"]
+
+    manual_tunnel = await hass.config_entries.flow.async_configure(
+        tunnel_flow["flow_id"],
+        {CONF_KNX_GATEWAY: str(gateway)},
+    )
+    await hass.async_block_till_done()
+    assert manual_tunnel["type"] == RESULT_TYPE_FORM
+    assert manual_tunnel["step_id"] == "manual_tunnel"
+
+    with patch(
+        "homeassistant.components.knx.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        manuel_tunnel_flow = await hass.config_entries.flow.async_configure(
+            manual_tunnel["flow_id"],
+            {
+                CONF_HOST: "192.168.0.1",
+                CONF_PORT: 3675,
+            },
+        )
+        assert manuel_tunnel_flow["data"] == {
```

Comment:
We should block till done when creating the entry to make sure we're still in the patch context manager until all tasks are done.

Reply:
Please always assert the result type.